Fixes for making entire video card clickable and enhancing course card display

carddev81 opened the pull request:
## Description of the change

- #673 
    - Modified CSS to fix the position of the progress bar at the bottom of card.
    - Added javascript and an html element to handle displaying the browser tooltip when course title is truncated
    - Modified javascript date formatter to display 2 digits for day rather than a numeric to align with MM/DD/YYYY format
- #679 
    - Moved onClick handler to parent div HTML element of the Video Card
- **Related issues**:  Closes #673, Closes #679.

## Screenshot(s)
![image](https://github.com/user-attachments/assets/db1cc80d-042e-4a72-ab55-59a8d1c317df)
![image](https://github.com/user-attachments/assets/67791f15-bc2e-42f3-b62f-9be7b336a116)

jtucholski reviewed (changes requested):
Video and changes to the catalog card look good. I can re-review when you've made changes on the ClampedText component.

carddev81 commented:
added clamped text component

carddev81 commented:
@jtucholski I was thinking that maybe I should go ahead and wrap all elements on this page that use Tailwind's CSS line-clamp-# style with the new ClampedText component.  Do you want me to go ahead and do this or just the title?  I also created some reference documentation here for the new component:  https://github.com/UnlockedLabs/UnlockEdv2/wiki/Component-Reference-Guide

jtucholski commented:
@carddev81 Something changed in this PR and the Progress bars are showing up on top of the overlay that happens when the menu is unpinned and showing. I don't see the issue over in `main`.

Perhaps the issue is in `EnrolledCourseCard.tsx`? 


 
![Screenshot 2025-01-27 124019](https://github.com/user-attachments/assets/cea78fb3-6ffe-4bf4-b61d-35bac072c83d)

![Screenshot 2025-01-27 123730](https://github.com/user-attachments/assets/d15935c7-bd4b-4864-82c9-8f55ac8d1a5d)

carddev81 commented:
I had added z index for the progress bar to display higher than the text within the card, didn't see that it caused that error. Checked it out and z index is not needed, and removed it altogether.
